[quant][graphmode] quantization support for aten::add

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35130 [quant][graphmode] Make interpolate/upsample work again
* #35142 [quant][graphmode] SwapDeQuant support prim::If
* #35141 [quant][graphmode][refactor] getGeneralOpTensorInputIndexes -> getGeneralOpTensorInputs
* #35135 [quant][graphmode][refactor] swapDeQuant takes block as arugment
* #35077 [quant][graphmode] Fold quantized prepacking ops
* #35073 [quant] Make conv2d_prepack and linear_prepack pure
* #34855 [quant][graphmode] quantization support for prim::CallFunction
* #34346 [quant][graphmode] Add quantization support for aten::cat
* #34345 [quant][graphmode] Add prim::ListConstruct to general op handling
* #34854 [quant][graphmode] quantization support for view, transpose, contiguos
* #34806 [quant][graphmode] quantization support for aten::chunk
* #34807 [quant][graphmode] quantization support for prim::ListUnpack
* #34804 [quant][graphmode] Replicate quantize node for prim::If
* **#34572 [quant][graphmode] quantization support for aten::add**

Summary:
This is only for Tensor - Tensor add, we'll need to support Tensor - Scalar add in a different way, since that does not need observer
Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

jerryzh168 committed:
[quant][graphmode] quantization support for aten::add

Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar reviewed (approved):
LGTM, do we need test for scalar add as well?

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 commented:
> LGTM, do we need test for scalar add as well?

actually we don't support scalar add right now, will update in the summary

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:
This adds support to Tensor add, we don't support scalar add right now, will support that later

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:
This adds support to Tensor add, we don't support scalar add right now, will support that later

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:
This adds support to Tensor add, we don't support scalar add right now, will support that later

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:
This adds support to Tensor add, we don't support scalar add right now, will support that later

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:
This adds support to Tensor add, we don't support scalar add right now, will support that later

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:
This adds support to Tensor add, we don't support scalar add right now, will support that later

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:
This adds support to Tensor add, we don't support scalar add right now, will support that later

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:
This adds support to Tensor add, we don't support scalar add right now, will support that later

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::add"


Summary:
This is only for Tensor - Tensor add, we'll need to support Tensor - Scalar add in a different way, since that does not need observer
Test Plan:
python test/test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20519607](https://our.internmc.facebook.com/intern/diff/D20519607)

[ghstack-poisoned]
